src/core/Akka/Actor/ChildrenContainer/Internal/ChildrenContainerBase.cs

```diff
@@ -30,25 +54,29 @@ protected ChildrenContainerBase(IImmutableDictionary<string, IChildStats> childr
         public abstract IChildrenContainer ShallDie(IActorRef actor);
         public abstract IChildrenContainer Unreserve(string name);
 
-        public IReadOnlyList<IInternalActorRef> Children
+        public IReadOnlyCollection<IInternalActorRef> Children
         {
             get
             {
-                return (from stat in InternalChildren.Values
-                        let childRestartStats = stat as ChildRestartStats
-                        where childRestartStats != null
-                        select childRestartStats.Child).ToList();
+                var children = from stat in InternalChildren.Values
+                               let childRestartStats = stat as ChildRestartStats
+                               where childRestartStats != null
+                               select childRestartStats.Child;
+
+                return new WrappedReadonlyCollection<IInternalActorRef>(children, InternalChildren.Count);
             }
         }
 
-        public IReadOnlyList<ChildRestartStats> Stats
+        public IReadOnlyCollection<ChildRestartStats> Stats
         {
             get
             {
-                return (from stat in InternalChildren.Values
+                var children = from stat in InternalChildren.Values
```

Comment:
same here. I find:
`return InternalChildren.Values.OfType<ChildRestartStats>();`
a little easier to read